Cut 2.26.0-rc.0 release. (#8640)
prometheus/prometheus: CHANGELOG.md

## 2.26.0-rc.0 / 2021-03-24

Prometheus is now built and supporting Go 1.16 (#8544). This reverts the memory release pattern added in Go 1.12. This makes common RSS usage metrics showing more accurate number for actual memory used by Prometheus. You can read more details [here](https://www.bwplotka.dev/2019/golang-memory-monitoring/).

Note that from this release Prometheus is using Alertmanager v2 by default.

* [CHANGE] Alerting: Using Alertmanager v2 API by default. #8626
* [CHANGE] Prometheus/Promtool: As agreed on dev summit, binaries are now printing help and usage to stdout instead of stderr. #8542
* [FEATURE] Remote: Add support for AWS SigV4 auth method for remote_write. #8509
* [FEATURE] Scaleway Discovery: Add Scaleway Service Discovery. #8555

> * [FEATURE] PromQL: Allow negative offsets. Behind `--enable-feature=promql-negative-offset` flag. #8555

Reviewer: Wrong PR or issue link. 8555 is about [FEATURE] Scaleway Discovery: Add Scaleway Service Discovery.

Reviewer: correct PR is https://github.com/prometheus/prometheus/pull/8487

* [FEATURE] **experimental** Exemplars: Add in-memory storage for exemplars. Behind `--enable-feature=exemplar-storage` flag. #6635
* [FEATURE] UI: Add advanced auto-completion, syntax highlighting and linting to graph page query input. #8634
* [ENHANCEMENT] Digital Ocean Discovery: Add `__meta_digitalocean_image` label. #8497

> * [ENHANCEMENT] Remote: Allow configuring retries on rate-limiting for remote_write. See [configuration docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) for details. #8477

Reviewer 2: Actually, the enhancement is #8237. This release made it disabled by default.

`* [ENHANCEMENT] Remote: Optionally retry 429 http responses for remote_write.`

## 2.26.0-rc.0 / 2021-03-24

> Notable changes:

Reviewer 2: The only changes that should stand up are the go 1.16 change and the move to alertmanager api v2.

> * Prometheus UI helps you to write correct PromQL queries, with auto-completion, syntax highlightning and linting. 

Reviewer 3: ```suggestion
* Experimental query editor UI helps you to write correct PromQL queries, with auto-completion, syntax highlighting and linting. 
```

> * Experimental query editor UI helps you to write correct PromQL queries, with auto-completion, syntax highlighting and linting. 

Reviewer 2: This should be a feature, like others.

> * Users replicating metrics via remote_write to AWS backends does not require extra sidecar, Prometheus now supports that method natively.

Reviewer 3: ```suggestion
* Native support for AWS authentication for `remote_write` to Amazon Managed Service for Prometheus. This eliminates the need for a sidecar.
```

> * Native support for AWS authentication for `remote_write` to Amazon Managed Service for Prometheus. This eliminates the need for a sidecar.

Reviewer 2: This is not notable, it is just a feature. The AWS service is not even GA. It feels like we promote a commercial service and I don't agree with this.

* If you enable `--enable-feature=exemplar-storage`, Prometheus will hold certain amount of scraped OpenMetrics' Exemplars in circular memory. It will also expose `/api/v1/query_exemplars` querying HTTP API.

> * If you enable `--enable-feature=promql-negative-offset`, PromQL will support negative offset in queries eg. `sum_over_time(metric[1h:1m] offset -5m)`.

Reviewer 3: ```suggestion
* PromQL will support negative offset in queries via feature flag `--enable-feature=promql-negative-offset`. (eg. `sum_over_time(metric[1h:1m] offset -5m)`)
```

> * If you enable `--enable-feature=exemplar-storage`, Prometheus will hold certain amount of scraped OpenMetrics' Exemplars in circular memory. It will also expose `/api/v1/query_exemplars` querying HTTP API.
> * If you enable `--enable-feature=promql-negative-offset`, PromQL will support negative offset in queries eg. `sum_over_time(metric[1h:1m] offset -5m)`.

Reviewer 4: Might be useful to link to the docs here. Also, it's a comma separated list, the way this is written reads a bit like both can be flags on their own, which might confuse people, but not sure there's a better way to write it.

```suggestion
* Two experimental features have been added, these are [disabled by default and may change in the future](https://prometheus.io/docs/prometheus/latest/disabled_features/):
    * If you enable `--enable-feature=exemplar-storage`, Prometheus will hold certain amount of scraped OpenMetrics' Exemplars in circular memory. It will also expose `/api/v1/query_exemplars` querying HTTP API.
    * If you enable `--enable-feature=promql-negative-offset`, PromQL will support negative offset in queries eg. `sum_over_time(metric[1h:1m] offset -5m)`.
```

Reviewer 4: Oh multiple flags do work, nevermind...

>   * If you enable `--enable-feature=exemplar-storage`, Prometheus will hold certain amount of scraped OpenMetrics' Exemplars in circular memory. It will also expose `/api/v1/query_exemplars` querying HTTP API.

Reviewer 2: This should only be covered below and in the announcement.

>   * PromQL support for negative offset in queries via feature flag `--enable-feature=promql-negative-offset`. (eg. `sum_over_time(metric[1h:1m] offset -5m)`)

Reviewer 2: This should only be covered below and in the announcement.

* This release changed the main development branch to `main`.

* [CHANGE] Alerting: Using Alertmanager v2 API by default. #8626
* [FEATURE] Remote: Add support for AWS SigV4 auth method for remote_write. #8509
* [FEATURE] Scaleway Discovery: Add Scaleway Service Discovery. #8555
* [FEATURE] PromQL: Allow negative offsets. Behind `--enable-feature=promql-negative-offset` flag. #8555
* [FEATURE] **experimental** Exemplars: Add in-memory storage for exemplars. Behind `--enable-feature=exemplar-storage` flag. #6635
* [FEATURE] UI: Add advanced auto-completion, syntax highlighting and linting to graph page query input. #8634
* [ENHANCEMENT] Digital Ocean Discovery: Add `__meta_digitalocean_image` label. #8497
* [ENHANCEMENT] Remote: Allow configuring retries on rate-limiting for remote_write. See [configuration docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) for details. #8477

> * [ENHANCEMENT] Authorization: Add support for specifying type of credentials with `Bearer` by default. #8512

Reviewer 2: ```suggestion
* [ENHANCEMENT] Authorization: Add support for specifying type of credentials with `Bearer` by default. #8512
```
 Scrape: Add support for specifying type of Authorization header credentials with `Bearer` by default

> * [ENHANCEMENT] Prometheus/Promtool: Binaries are now printing help and usage to stdout instead of stderr. #8542

Reviewer 2: ```suggestion
* [CHANGE] Prometheus/Promtool: Binaries are now printing help and usage to stdout instead of stderr. #8542
```

As agreed in dev summit.

Reviewer 2: I meant that it was agreed on dev summit to mark as a change, it was not needed to write it as is here ;)

> * [ENHANCEMENT] Remote: Allow configuring custom headers for remote_read. See [configuration docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_read) for details. #8516

Reviewer 2: it is not needed to point to the docs, because the location is not surprising.

Author: But it's not harmful too?

Author: I will leave it, I don't see reason why not being more explicit for users.

* [ENHANCEMENT] Scrape: Add `follow_redirects` option to scrape configuration. #8546
* [ENHANCEMENT] UI: Hitting Enter now triggers new query. #8581
* [ENHANCEMENT] UI: Better handling of long rule names on `/rules` page. #8608
* [ENHANCEMENT] UI: Better handling of long labels names on `/targets` page. #8609
* [BUGFIX] TSDB: Eager deletion of removable blocks on every compaction, saving disk peak space usage. #8007
* [BUGFIX] PromQL: Fix parser support for special characters like`炬`. #8517

> * [BUGFIX] TSDB: Fix rare case of duplicated sample. #8591

Reviewer 2: This should not be there, it was released in 2.25.1.

Author: how can I tell? 
![image](https://user-images.githubusercontent.com/6950331/112333387-77635d00-8cb2-11eb-80c5-5418d1e65caf.png)

Author: Anyway, removed, just was curious.

Reviewer 2: You can't,but we are a team :) As 2.25 shepherd I cherry picked this for a release.
